js/calendar.js

```diff
@@ -83,3 +91,153 @@ function refreshCalendar(nextState) {
+		var startDay = eventData.start.getDate();
+		var endDay = eventData.end.getDate();
+		var startMonth = eventData.start.getMonth();
+		var endMonth = eventData.end.getMonth();
+		var startYear = eventData.start.getFullYear()
+		var endYear = eventData.end.getFullYear();
+		const months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+		var startDate = "";
+		var endDate = "";
+		if (startYear != endYear) {
+			// years differ - "dd MMM YYYY - dd MMM YYYY"
+			startDate = startDay + space + months[startMonth] + space + startYear;
+			endDate = endDay + space + months[endMonth] + space + endYear;
+			if (!allDay) {
+				// "dd MMM YYYY HH:MM - dd MMM YYYY HH:MM"
+				startDate += space + startHour + colon + startMinute;
+				endDate += space + startHour + colon + startMinute;
+			}
+		} else if (startMonth != endMonth) {
+			// same year, different month - "dd MMM - dd MMM"
+			startDate = startDay + space + months[startMonth];
+			endDate = endDay + space + months[endMonth];
+			if (!allDay) {
+				// "dd MMM HH:MM - dd MMM HH:MM"
+				startDate += space + startHour + colon + startMinute;
+				endDate += space + startHour + colon + startMinute;
+			}
+		} else if (startDay != endDay) {
+			// same year, same month, different day - "dd - dd MMM YYYY"
+			if (allDay) {
+				startDate = startDay;
+				endDate = endDay + space + months[endMonth] + space + endYear;
+			} else {
+				// "dd MMMM YYYY HH:MM - dd MMM YYYY HH:MM"
+				startDate = startDay + space + months[startMonth] + space + startHour + colon + endMinute;
+				endDate = endDay + space + months[endMonth] + space + endHour + colon + endMinute;
+			}
+		} else {
+			// same day - "dd MMM HH:MM - HH:MM"
+			startDate = startDay + space + months[startMonth] + space + startHour + colon + startMinute;
+			endDate = endHour + colon + endMinute;
+
+		}
+		date = startDate + dash + endDate;
+
+		//	location
+		location = eventDataExtended.location;
+
+		// description 
+		description = parseDescription(eventDataExtended.description);
+
+		// Facebook event button
+		let eventFacebookLinkElement = document.getElementById("facebook-link");
+		let link = parseFacebookLink(eventDataExtended.description);
+		if (link) {
+			eventFacebookLinkElement.onclick = () => {window.open(link)};
+			eventFacebookLinkElement.style.display = "inline";
+		} else {
+			// catches null regex matches
+			eventFacebookLinkElement.style.display = "none";
+		}
+	
+
+		// set text
+		document.getElementById("event-text-title").textContent = title;
+		document.getElementById("event-text-date").textContent = date;
+		document.getElementById("event-text-location").textContent = location;
+		var descStr = "";
+		for (line in description) {
+			descStr += description[line].toString() + "<br>";
+		}
+		document.getElementById("event-text-description").innerHTML = descStr; //innerhtml since description is already string
```

Comment:
Hey, so let's not use this if possible. The input strings come from the calendar, but probably shouldn't be converted directly into html, as that's a possible vector for an XSS attack (which technically is only accessible by CSS committee, but still... the security of one part shouldn't dictate the security of all the parts). The protections below aren't sufficient because we convert the data into a string, and then right back into html, which is dangerous.

Instead we should do the above, where we modify text content directly. For the newlines, instead of using `<br>`, I think we can get away with using `'\n'`? If that doesn't work, then maybe try changing the element type to `pre` might.

Reply:
Worked when using `pre`. For future reference, didn't work when using only `textContent` with `\n`

Reply:
You could use the .innerText attribute to set the text. ```\n``` will work as the line break there.